lib/logstash/codecs/protobuf.rb

```diff
@@ -0,0 +1,189 @@
+# encoding: utf-8
+require 'logstash/codecs/base'
+require 'logstash/util/charset'
+require 'protocol_buffers' # https://github.com/codekitchen/ruby-protocol-buffers
+
+class LogStash::Codecs::Protobuf < LogStash::Codecs::Base
```

Comment:
I'm sorry I can't help because I've never generated the plugin doc, and I don't know if there is a way to preview it. Maybe @suyograo has a way to do that?

As Suyog suggests, make sure that you provide documentation above the class definition to describe the plugin and how it's used. Any valid asciidoc formatting should be supported. You can check out some of the existing plugins for examples of what's supported. As you can see here, it's possible to add formatting for headers, bulleted lists, etc: https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch.rb